src/shared_queue.h

```diff
@@ -50,15 +47,14 @@ class SharedQueue {
 
   // TODO: use std::shared_mutex in C++17.
   std::mutex mutex_;
-  std::map<uint32_t, Queue> queues_;
   uint32_t next_queue_token_ = 1;
-  struct pair_hash {
-    template <class T1, class T2> std::size_t operator()(const std::pair<T1, T2> &pair) const {
-      return std::hash<T1>()(pair.first) ^ std::hash<T2>()(pair.second);
-    }
-  };
-  std::unordered_map<std::pair<std::string, std::string>, uint32_t, pair_hash> queue_tokens_;
-  std::unordered_set<uint32_t> queue_token_set_;
+
+  // vm_id -> queue keys
+  std::unordered_map<std::string, std::unordered_set<std::size_t>> vm_queue_keys_;
+  // queue key -> token
+  std::unordered_map<std::size_t, uint32_t> queue_tokens_;
+  // token -> queue
+  std::map<uint32_t, Queue> queues_;
```

Comment:
Is there any reason why this is `std::map` instead of `std::ordered_map`?

Reply:
hah, no reason. just a mistake. give me a sec